src/controlplane/client/connector.go

```diff
@@ -0,0 +1,274 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/nri-kubernetes/v2/internal/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+
+	"github.com/newrelic/nri-kubernetes/v2/src/client"
+)
+
+const (
+	DefaultTimout      = 5000 * time.Millisecond
+	defaultMetricsPath = "/metrics"
+	plainAuth          = ""
+	mTLSAuth           = "mTLS"
+	bearerAuth         = "bearer"
+)
+
+// Connector provides an interface to retrieve []connParams to connect to a Control Plane instance.
+type Connector interface {
+	Connect() (*connParams, error)
+}
+
+type defaultConnector struct {
+	// TODO: Use a non-sdk logger
+	logger          log.Logger
+	kc              kubernetes.Interface
+	inClusterConfig *rest.Config
+	endpoints       []config.Endpoint
+}
+
+// DefaultConnector returns a defaultConnector that probes all endpoints in the list and return the first responding status OK.
+func DefaultConnector(endpoints []config.Endpoint, kc kubernetes.Interface, inClusterConfig *rest.Config, logger log.Logger) (Connector, error) {
+	if err := validateEndpointConfig(endpoints); err != nil {
+		return nil, fmt.Errorf("validating endpoints config: %w", err)
+	}
+
+	return &defaultConnector{
+		logger:          logger,
+		inClusterConfig: inClusterConfig,
+		kc:              kc,
+		endpoints:       endpoints,
+	}, nil
+}
+
+// Connect iterates over the endpoints list and returns the connection parameters of the
+// first endpoint that respond Status OK.
+func (dp *defaultConnector) Connect() (*connParams, error) {
+	for _, e := range dp.endpoints {
+
```

Comment:
I would add a log.debug specifying what we are checking

Reply:
and in general more logs so that in case of weird behaviours we can follow expactly in debug mode what is happening